src/track/beats.cpp

```diff
@@ -329,15 +307,65 @@ Bpm BeatsInternal::getBpm() const {
 bool BeatsInternal::isValid() const {
     return m_iSampleRate > 0 && !m_beats.empty();
 }
+void BeatsInternal::setSampleRate(int sampleRate) {
+    m_iSampleRate = sampleRate;
+    updateBpm();
+}
 BeatsInternal::BeatsInternal()
         : m_iSampleRate(0), m_dDurationSeconds(0) {
 }
+
+BeatsInternal::BeatsInternal(const QByteArray& byteArray)
+        : BeatsInternal() {
+    track::io::Beats beatsProto;
+    if (!beatsProto.ParseFromArray(byteArray.constData(), byteArray.size())) {
+        qDebug() << "ERROR: Could not parse kBeatMap from QByteArray of size"
+                 << byteArray.size();
+    }
+    for (int i = 0; i < beatsProto.beat_size(); ++i) {
+        const track::io::Beat& beat = beatsProto.beat(i);
+        m_beats.append(beat);
+    }
+
+    DEBUG_ASSERT(beatsProto.time_signature_markers_size() > 0);
+    for (int i = 0; i < beatsProto.time_signature_markers_size(); ++i) {
+        const track::io::TimeSignatureMarker& timeSignatureMarker =
+                beatsProto.time_signature_markers(i);
+        m_timeSignatureMarkers.append(timeSignatureMarker);
+    }
+    updateBpm();
+}
+
+BeatsInternal::BeatsInternal(const QVector<FramePos>& beats)
```

Comment:
We also need the downbeats, phrases, sections, and time signatures from the analyzer. For now, add these arguments to the constructor signature and add these assuming defaults.